repos/system_upgrade/common/actors/scancpu/libraries/scancpu.py

```diff
@@ -137,4 +142,10 @@ def process():
     api.produce(*_find_deprecation_data_entries(lscpu))
     # Backwards compatibility
     machine_type = lscpu.get('Machine type')
```

Comment:
Just wondering if we can end up with non-int here? If yes conversion at L148 could fail

Reply:
I'm unsure. This is the way I've found it. Looking at the source of ``lscpu``, it seems to be getting the "Machine type" field from the "Type" field in the``/proc/sysinfo`` file [ref. 1][1] . In the kernel source for the s390x architecture, this field is printed and stored as a string [ref. 2][2], [ref. 3][3]. This information is generated from reading the result of the STSI instruction of s390x. However, I was not able to find a reference that would specify what values can be produced there. So I guess it is possible hmm

[1]: https://github.com/util-linux/util-linux/blob/50d216859f7ec6a5b200a8fb90c9a0870d12e018/sys-utils/lscpu-cputype.c#L741
[2]: https://github.com/torvalds/linux/blob/94710cac0ef4ee177a63b5227664b38c95bbf703/arch/s390/kernel/sysinfo.c#L91
[3]: https://github.com/torvalds/linux/blob/94710cac0ef4ee177a63b5227664b38c95bbf703/arch/s390/include/asm/sysinfo.h#L27

Reply:
No worries! I've been checking the lscpu source for the stuff around flags as well so it was pretty quick. I've made one last quick search for the description of the STSI instruction and found it described in [z/Architecture Principles of Operation, page 114][1] with

> Type: Word 12 contains the four-character (0-9) EBCDIC type number of the configuration. (This is called the machine-type number in the definition of STORE CPU ID.)

So the only problem is when the type starts with leading zeros, which of course will not be shown when converted to int, but I think this is not a problem and I'll keep it as is :)

[1]: https://geraldine.fjfi.cvut.cz/~oberhuber/data/mainframe/redbooks/pmf/z-architecture-principles-of-operations.pdf